containerd-shim-v2/service.go

```diff
@@ -161,6 +208,14 @@ func newCommand(ctx context.Context, containerdBinary, id, containerdAddress str
 // StartShim willl start a kata shimv2 daemon which will implemented the
 // ShimV2 APIs such as create/start/update etc containers.
 func (s *service) StartShim(ctx context.Context, id, containerdBinary, containerdAddress string) (string, error) {
+	// Stop tracing because a new tracer will be created when New is called again after StartShim
+	defer katautils.StopTracing(s.ctx)
```

Comment:
@jodh-intel I'm not sure I understand this. I could call `StopTracing()` before the tracer is created in `New()`, but `s.tracer` doesn't exist until halfway through the function. A new context is used every time `New()` is called. `New()` is called twice per container, initially followed by `StartShim()`, so there is one context used between those two, and a different context for the second `New()` and subsequent functions.

Reply:
@jodh-intel I thought about this some more and tracing can't be stopped earlier in `New()` because stopping tracing depends on using a specific context, which isn't available since a new one is used in that function. It is possible I'm missing functions in between `New()` and `StartShim()`, but `StartShim()` is the only function that I've seen called after `New()` with the same context where we can end the first trace.